File: homework/home4.1/src/Pizza.java
import java.util.List;

public class Pizza {
    private final List<Ingredient> ingredients;

    public Pizza(List<Ingredient> ingredients) {
        this.ingredients = ingredients;
    }

    public double getTotalCost() {
        return ingredients.stream().mapToDouble(Ingredient::getCost).sum();
    }

    public void displayPizza() {
        System.out.println("Your pizza ingredients:");
        ingredients.forEach(ingredient ->
                System.out.printf("- %s ($%.2f)%n", ingredient.getName(), ingredient.getCost()));
        System.out.printf("Total cost: $%.2f%n", getTotalCost());
    }
}

File: classwork/classwork4.1/Pizza.cpp
#pragma once

#include <vector>
#include <string>
#include <iostream>
#include <numeric>
#include "Ingredient.cpp"

class Pizza {
private:
    std::vector<Ingredient> ingredients;

public:
    Pizza(const std::vector<Ingredient>& _ingredients) : ingredients(_ingredients) {}

    double getTotalCost() const {
        return std::accumulate(ingredients.begin(), ingredients.end(), 0.0,
                               [](double sum, const Ingredient& ingredient) {
                                   return sum + ingredient.getCost();
                               });
    }

    void displayPizza() const {
        std::println("Your pizza ingredients:");
        for (const auto& ingredient : selectedIngredients) {
            std::println("- {} (${})", ingredient.getName(), ingredient.getCost());
        }
        std::println("Total cost: ${}", getTotalCost());
    }
};
